packages/core-data/src/reducer.js

```diff
@@ -623,6 +623,18 @@ export function defaultTemplates( state = {}, action ) {
 	return state;
 }
 
+export function linkSuggestions( state = {}, action ) {
+	switch ( action.type ) {
+		case 'RECEIVE_LINK_SUGGESTIONS':
+			return {
+				...state,
+				[ action.search ]: action.suggestions,
```

Comment:
I don't think this indexing strategy is good enough. A search is a simple string but the API allows for searching by different entities such as `post`, `page`...etc. So if we just index by the search you could accidentally receives results for the wrong content type.